pkg/batchpertrace/batchpertrace.go

```diff
@@ -63,3 +63,51 @@ func Split(batch pdata.Traces) []pdata.Traces {
 
 	return result
 }
+
+// SplitLogs returns one pdata.Logs for each trace in the given pdata.Logs input. Each of the resulting pdata.Logs contains exactly one trace.
+func SplitLogs(batch pdata.Logs) []pdata.Logs {
+	// for each log in the resource logs, we group them into batches of rl/ill/traceID.
+	// if the same traceID exists in different ill, they land in different batches.
+	var result []pdata.Logs
+
+	for i := 0; i < batch.ResourceLogs().Len(); i++ {
+		rs := batch.ResourceLogs().At(i)
+
+		for j := 0; j < rs.InstrumentationLibraryLogs().Len(); j++ {
+			// the batches for this ILL
+			batches := map[[16]byte]pdata.ResourceLogs{}
```

Comment:
I think a pdata.TraceID can be used here now. It wasn't possible in the first implementation, IIRC.

Reply:
Yes because we were exposing as byte slice not array

Reply:
Fixed it both for `SplitLogs` and `SplitTraces` function.